### ISSUE TITLE
[Q] How to share chart setup between sweeps

### ISSUE THREAD

**Issue by davidgilbertson** [wandb/wandb#5564]
Hi, I've run a sweep, and created a few charts, changed the gradient for the default chart, and pinned a few columns to the left on the sweeps page of the web app. Now, I see that my range for one variable wasn't quite right so I run another sweep. But when I view that sweep, I'm back to the default screen (no pinned columns, scatter charts from the previous sweep not there).

In the last few days I've run 30+ sweeps with various tweaks and each time I'm having to re-add and re-configure charts and re-pin columns. Am I doing something dumb? Is there a better workflow?

**Comment by luisbergua** [comment]
Hi @davidgilbertson, thanks for writing in! So you'd like to be able to transfer the layout across different sweeps in the same project right? This isn't possible now so I can create a feature request and share internally, feel free to share any other details about your use case here!

**Comment by davidgilbertson** [comment]
I assume that sharing of charts between all sweeps would be very difficult. But at the very least, I think if a sweep runs, some charts are created, then another sweep runs _with the same metrics in the config_, then you could allow the user the copy across the UI config from the prior sweep. In short, if the next config is compatible with the previous charts, reuse the previous charts.

A real life example, I just started a sweep with, among other things, learning rate. I put the max at 1. But after a few runs I saw that the best results were really close to  lr=1. 
![image](https://github.com/wandb/wandb/assets/4443482/f488a503-ff06-4f12-84da-aa2f4921658c)

So rather than let this run for another hour or so, I decide that I should kill it, increase the max LR to 10, and start again. And of course that means re-creating the charts yet another time.

**Comment by luisbergua** [comment]
Hi @davidgilbertson, thanks a lot for sharing this feedback! I'll create the feature request and share it with our Product Team.

**Comment by markschoene** [comment]
This feature would be very welcome. Especially when running multiple sweeps in the same project with slight variations to the hyperparameters... The view option allows us to save a view of a single workspace. Why not make transfer of the view between sweeps in the same project possible?

**Comment by luisbergua** [comment]
Hey @markschoene, thanks for your feedback! I'll add you to the FR and share this internally

**Comment by umarbutler** [comment]
+1 I'd like to see this feature implemented as well. Not only for 'sweep runs with the same metrics in the config' as @davidgilbertson suggested, but for *all* sweeps.

I have run hundreds of sweeps this past week and it would have saved me a lot of time to not have to switch every panel to display f1 instead of loss, hide panels I don't want, add new panels I do want, etc...

**Comment by luisbergua** [comment]
Thanks for adding this extra feedback @umarbutler! I'll ensure to raise it with the team

**Comment by zurdaniel** [comment]
+1
I have faced the same issue. I would appreciate any updates on this.

**Comment by JCCPort** [comment]
+1
It's currently a very major slowdown for me

**Comment by thesebelik-MU** [comment]
Just discovered this thread trying to search for a way to share views between sweeps, thinking it must already be a feature. My current project involves the creation of a lot of individual sweeps and I feel like I spend more time re-creating the same exact tables and panels within each sweep (so I can effectively compare their results) than I do on actually training the models. This thread is now two years old, so I would love to hear some updates on this.

**Comment by luisbergua** [comment]
Hey everyone, thanks for the feedback! While this isn't fully possible yet, we have built an API to interact with workspaces programmatically (docs [here](https://docs.wandb.ai/models/track/workspaces#programmatically-create-workspaces)), which will be helpful here. If you save a view, you can then [copy](https://docs.wandb.ai/models/track/workspaces#copy-a-workspace-saved-view-to-another-workspace) it to other projects/workspaces to avoid manually recreating it. Please let me know if that would help!

**Comment by akriese** [comment]
Hi @luisbergua, I just tried out the API to copy the view of one sweep to another and it seems like URL validation does not allow sweep workspaces currently. It expects `https://wandb.ai/{entity}/{project}?nw=a0b1c2d3` while the sweep workspace url is of shape `https://wandb.ai/{entity}/{project}/sweeps/{sweep_id}?nw=a0b1c2d3`.

**Comment by luisbergua** [comment]
hey @akriese, thanks for getting back! yes that's correct, if you want to copy the dedicated sweep that's currently not possible via api, i have raised this with our product team and will post here as soon as i hear back from them
